services/api/src/utils/logging.js

```diff
@@ -1,22 +1,133 @@
-function timeStr() {
-  return new Date().toString();
+const pino = require('pino');
+const config = require('@bedrockio/config');
+
+const tracer = require('./tracer');
+const startTime = Symbol('startTime');
+
+function getGlobalContext() {
+  const currentSpan = tracer.getCurrentSpan();
+  if (!currentSpan) return null;
+  return currentSpan.context();
 }
 
-const logger = {
-  debug(...args) {
-    console.info(...[timeStr(), '-', 'DEBUG', '-'].concat(args));
-  },
-  info(...args) {
-    console.info(...[timeStr(), '-', 'INFO', '-'].concat(args));
+const parentLogger = pino({
+  messageKey: 'message',
+  formatters: {
+    level(_, level) {
+      if (level === 20) {
+        return { severity: 'debug' };
+      }
+      if (level === 30) {
+        return { severity: 'info' };
+      }
+      if (level === 40) {
+        return { severity: 'warning' };
+      }
+      if (level === 50) {
+        return { severity: 'error' };
+      }
+      if (level >= 60) {
+        return { severity: 'critical' };
+      }
+      return { severity: 'default' };
+    },
   },
-  warn(...args) {
-    console.info(...[timeStr(), '-', 'WARN', '-'].concat(args));
-  },
-  error(...args) {
-    console.info(...[timeStr(), '-', 'ERROR', '-'].concat(args));
+  base: null,
+  level: config.get('LOG_LEVEL'),
+  timestamp: pino.stdTimeFunctions.isoTime,
+  prettyPrint: process.env.NODE_ENV === 'dev' ? { messageKey: 'message' } : false,
```

Comment:
curious but you're piping `yarn start` to `pino-pretty` for the dev server... are both that and the above required to work? Seems like one or the other would be enough.

Reply:
gonna move it to the package.json as cli option e.g. `pino-pretty --messageKey=message`